src/main/java/og_spipes/service/ScriptOntologyHelper.java

```diff
@@ -33,13 +33,23 @@ public Map<URI, File> moduleFile(Set<URI> modules){
             String baseURI = OntologyDao.getOntologyUri(ff);
             if(!baseURI.isBlank()){
                 String importName = baseURI + "/";
-                List<String> okModules = modules.stream().map(URI::toString)
+                List<String> okModules = modules.stream()
+                        .map(uri -> {
+                            if (uri.getFragment() != null) {
+                                return uri.getScheme() + "://" + uri.getHost() + uri.getPath() + "/" + uri.getFragment();
+                            } else {
+                                return uri.toString();
+                            }
+                        })
+                        .filter(x -> x.startsWith(importName))
                         .filter(x -> !x.replace(importName, "").contains("/"))
                         .collect(Collectors.toList());
 
                 Set<String> subjects = OntologyDao.getSubjects(ff);
-                for(String module : okModules){
-                    if(subjects.stream().anyMatch(x -> x.equals(module))){
+                for (String module : okModules) {
+                    String hashForm = module.contains("/") ? replaceLast(module, "/", "#") : module;
```

Comment:
This is third time that we do the same as normalizeUri(URI uri) right ? and why we even call it normalizeURI ?